zipline/_protocol.pyx

```diff
@@ -76,6 +79,289 @@ cdef class BarData:
             self.data_frequency
         )
 
+    def spot_value(self, assets, fields):
+        """
+        Returns the spot value of the given assets for the given fields
+        at the current simulation time.  Spot values are the as-traded price
+        and are not adjusted for any events like splits or dividends.
+
+        Parameters
+        ----------
+        assets : Asset or iterable of Assets
+
+        fields : string or iterable of strings.  Valid values are: "price",
+            "last_traded", "open", "high", "low", "close", "volume"
+
+        Returns
+        -------
+        Float, pandas Series, or pandas DataFrame.  See notes
+        below.
+
+        Notes
+        -----
+        If a single asset and a single field are passed in, a scalar float
+        value is returned.
+
+        If a single asset and a list of fields are passed in, a pandas Series
+        is returned whose indices are the fields, and whose values are scalar
+        values for this asset for each field.
+
+        If a list of assets and a single field are passed in, a pandas Series
+        is returned whose indices are the assets, and whose values are scalar
+        values for each asset for the given field.
+
+        If a list of assets and a list of fields are passed in, a pandas
+        DataFrame is returned, indexed by asset.  The columns are the requested
+        fields, filled with the scalar values for each asset for each field.
+
+        "price" returns the last known close price of the asset.  If there is
+        no last known value, NaN is returned.  If a value is found, and we
+        had to cross an adjustment boundary (split, dividend, etc) to get it,
+        the value is adjusted before being returned.
+
+        "last_traded" returns the date of the last trade event of the asset,
+        even if the asset has stopped trading. If there is no last known value,
+        NaT is returned.
+
+        "volume" returns the trade volume for the current simulation time.  If
+        there is no trade this minute, 0 is returned.
+
+        "open", "high", "low", and "close" return the relevant information for
+        the current trade bar.  If there is no current trade bar, NaN is
+        returned.
+        """
+        if isinstance(assets, Asset):
+            if isinstance(fields, str):
+                # return scalar value
+                return self.data_portal.get_spot_value(
+                    assets,
+                    fields,
+                    self.simulation_dt_func(),
+                    self.data_frequency
+                )
+            else:
+                # assume fields is iterable
+                # return a Series indexed by field
+                return pd.Series(data={
+                    field: self.data_portal.get_spot_value(
+                        assets,
+                        field,
```

Comment:
We should make (but not required for this PR) `get_spot_value` accept the iterable as well.